py/server/modules/snapshot_methods.py

```diff
@@ -0,0 +1,145 @@
+
+    prop_id = deserialized_proposal_dict['Proposal']['ID']
+    from_time = deserialized_proposal_dict['Proposal']['from']
+    to_time = deserialized_proposal_dict['Proposal']['to']
+    author = deserialized_proposal_dict['Proposal']['Author']
+    chain_tip_height = deserialized_proposal_dict['Proposal']['block_height']
+    chain_tip_hash = deserialized_proposal_dict['Proposal']['block_hash']
+
+    prop = VotingProposal(
+        in_id=prop_id,
+        bl_height=chain_tip_height,
+        bl_hash=chain_tip_hash,
+        from_time=from_time,
+        to_time=to_time,
+        author=author)
+
+    proposal_dict[prop_id] = prop
+    active_proposal = prop
+
+# used only when mocking nsc
+def add_ownership_entry(data_json):
+    try:
+        new_owner = data_json['owner']
+        new_addr = data_json['address']
+        base58.b58decode_check(new_addr)
+
+    except KeyError as e:
+        response = {
+            "error": {
+                "code": 101,
+                "description": "Can not add ownership",
+                "detail": "invalid json request data, could not find field: " + str(e)
+            }
+        }
+    except ValueError as e:
+        response = {
+            "error": {
+                "code": 102,
+                "description": "Can not add ownership",
+                "detail": "address not valid: " + str(e)
+            }
+        }
+    else:
+        if len(new_owner) != 42 or not all(c in string.hexdigits for c in new_owner[2:]):
+            response = {
+                "error": {
+                    "code": 103,
+                    "description": "Can not add ownership",
+                    "detail": "Invalid owner string length != 42 or not an hex string"
+                }
+            }
+        else:
+
+            if new_owner in MOCK_MC_ADDRESS_MAP.keys():
+                # this is actually a reference
+                addresses = MOCK_MC_ADDRESS_MAP[data_json['owner']]
+                found = False
+                for entry in addresses:
+                    if entry == new_addr:
+                        found = True
+                        break
+                if not found:
+                    addresses.append(data_json['address'])
+                    response = {"status": "Ok"}
+                else:
+                    response = {
+                        "error": {
+                            "code": 104,
+                            "description": "Can not add ownership",
+                            "detail": "Ownership already set"
+                        }
+                    }
+            else:
+                MOCK_MC_ADDRESS_MAP[new_owner] = [new_addr]
+                response = {"status": "Ok"}
+
+    return response
+
+
+def get_mc_address_map(sc_address=None):
+    if mock_nsc:
+        return MOCK_MC_ADDRESS_MAP
+    else:
+        return get_nsc_ownerships(sc_address)
+
+def get_owner_sc_addr_list():
+    if mock_nsc:
+        return MOCK_OWNER_SC_ADDR_LIST
```

Comment:
Is it needed to check again for `mock_nsc` because it's already done in zen_balancer before it's called?

Reply:
You are right, I removed the check from the caller